compiler/java/src/main/java/com/c/lexer/UnknownCharException.java

```diff
@@ -0,0 +1,9 @@
+package com.c.lexer;
+
+public class UnknownCharException extends Exception {
```

Comment:
See my review in the lexer. You could print the code position here.

Reply:
done